02 Algorithm Reference/03 Handling Data/07 QuoteBars.html

```diff
@@ -0,0 +1,42 @@
+<meta name="tag" content="data event handlers"/>
+<meta name="tag" content="quotebars event"/>
+
+<p>
+    $[QuoteBar,T:QuantConnect.Data.Market.QuoteBar] provides Open, High, Low, Close, Bid, Ask, LastBidSize and LastAskSize properties for a given period of time. The Bid and the Ask properties are $[Bar,T:QuantConnect.Data.Market.Bar] objects that contains Open, High, Low and Close. The QuoteBar Open, High, Low and Close properties values are the mean of the respective Bid and Ask property. QuoteBars are supported only for Forex securities and futures contracts.
+</p>
+<div class="section-example-container">
+<pre class="python">
+    self.AddForex('EURUSD', Resolution.Hour) # Subscribe to hourly QuoteBars
+        
+ def OnData(self, data):
+    ## You can access the EURUSD QuoteBar by directly by subsetting the slice object
+    ## Note that all price attributes are averages of their repective bid and ask attributes
+    fxOpen  = data['EURUSD'].Open          ## Market Open FX Rate
+    fxClose = data['EURUSD'].Close         ## Market Close FX Rate
+    
+    ## If you are subscribed to more than one Forex or futures data stream then your can
+    ## access the QuoteBar dictionary and then subset this for your desired Forex symbol
+    fxQuoteBars = data.QuoteBars
+    eurusd      = fxQuoteBars['EURUSD']
+    fxOpen      = eurusd.Open          ## Market Open FX Rate
+    fxClose     = eurusd.Close         ## Market Close FX Rate
+            
+</pre>
+</div>
+<p>
+    With a specific QuoteBar you can also access Bid and Ask Bars for the same security. These Bars provide information specific to the Bid and Ask side of Forex and futures secuirites, whereas QuoteBars use the mean of the Open, High, Low, and Close attributes of the Bid and Ask Bars to extrapolte values.
```

Comment:
Lets clarify this with an example:

<p>
The quotebar.Open, High, Low and Close properties are the midpoint of the Bid-Ask spread at that moment of time. 
</p>

<p>
The quote bar also contains two sub-objects; <code>quotebar.Bid</code> and <code>quotebar.Ask</code>. These Bid and Ask bars have Open, High, Low and close properties. e.g. <code>quotebar.Bid.Open</code>.
</p>

Reply:
```suggestion
    With a specific QuoteBar you can also access Bid and Ask Bars for the same security. These Bars provide information specific to the Bid and Ask side of Forex and futures secuirites, whereas QuoteBars use the mean of the Open, High, Low, and Close attributes of the Bid and Ask Bars to extrapolte values.
```